tests/metatype/ttypetraits.nim

```diff
@@ -231,3 +231,19 @@ block genericHead:
   type Bar = object
   doAssert not compiles(genericHead(Bar))
   # doAssert seq[int].genericHead is seq
+
+block: # elementType
+  iterator myiter(n: int): auto =
```

Comment:
`elementType` should work fine over sequences, arrays, strings, etc. These could be covered in the test suite as well

Reply:
all these already are covered in test suite, see lines below

Reply:
In our libraries, I quite often use it over types as well (not just over expressions that return regular values):
https://github.com/status-im/nim-beacon-chain/blob/master/beacon_chain/ssz/types.nim#L302

Reply:
I'm not necessarily against a future PR `template ElementType(T: typedesc): typedesc` but 
`elementType` (taking an untyped value) is the better approach IMO since it works with called iterators, see test suite.